markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java

```diff
@@ -275,29 +275,132 @@ public static int getOrderedListNumber(@NonNull String line) {
      *
      * @return the new cursor position
      */
-    public static int togglePunctuation(@NonNull Editable editable, int selectionStart, int selectionEnd, @NonNull String punctuation) {
+   public static int togglePunctuation(@NonNull Editable editable, int selectionStart, int selectionEnd, @NonNull String punctuation) {
         switch (punctuation) {
             case "**":
             case "__":
             case "*":
             case "_":
             case "~~": {
-                final boolean selectionIsSurroundedByPunctuation = selectionIsSurroundedByPunctuation(editable, selectionStart, selectionEnd, punctuation);
-                if (selectionIsSurroundedByPunctuation) {
-                    editable.delete(selectionEnd, selectionEnd + punctuation.length());
-                    editable.delete(selectionStart - punctuation.length(), selectionStart);
-                    return selectionEnd - punctuation.length();
-                } else {
-                    final int containedPunctuationCount = getContainedPunctuationCount(editable, selectionStart, selectionEnd, punctuation);
-                    if (containedPunctuationCount == 0) {
-                        editable.insert(selectionEnd, punctuation);
-                        editable.insert(selectionStart, punctuation);
-                        return selectionEnd + punctuation.length() * 2;
-                    } else if (containedPunctuationCount % 2 > 0) {
-                        return selectionEnd;
-                    } else {
-                        removeContainingPunctuation(editable, selectionStart, selectionEnd, punctuation);
-                        return selectionEnd - containedPunctuationCount * punctuation.length();
+                String temp=editable.toString();
```

Comment:
1. Please mark those variables as `final` to avoid modifying their state accidentally.
2. Please only declare those variables in the blocks where you actually need them, for example `punctuationTriple` can be moved down at least to the `else` cases.